packages/mds-utils/deep-pick-properties.ts

```diff
@@ -0,0 +1,78 @@
+import { DeepPick, DeepPickPath } from 'ts-deep-pick'
+
+/**
+ * deepPickProperties exists as a way to extract deeply nested properties from an object.
+ *
+ * @param entity The entity to pick from
+ * @param unparsedPaths Paths to pick from entity. SUPPORTED GRAMMAR: Key names, separated by '.' for property access. The type may say it supports otherwise, but this is all that's supported.
+ * @returns The deeply picked entity
+ * @example
+ * ```typescript
+ * // You can pass in a list of keys (even nested ones!) to extract those values from the type
+ * const x: { telemetry: { gps: { lat: 1 } }, event_type: 'foo' } = deepPickProperties({ telemetry: { gps: { lat: 1, lng: 2 } }, event_type: 'foo' }, [
+      'telemetry.gps.lat',
+      'event_type'
+    ])
+ * ```
+ * @example
+ * ```typescript
+ * // You can pass in no paths in order for deepPickProperties to act as the identity function.
+ * const y: { telemetry: { gps: { lat: 1, lng: 2 } }, event_type: 'foo' } = deepPickProperties({ telemetry: { gps: { lat: 1, lng: 2 } }, event_type: 'foo' })
+ */
+export const deepPickProperties = <T, P extends DeepPickPath<T>[] = ['*']>(
+  entity: T,
+  unparsedPaths?: P
+): P extends ['*'] ? DeepPick<T, '*'> : DeepPick<T, P[number]> => {
+  if (typeof unparsedPaths === 'undefined' || unparsedPaths.length === 0)
```

Comment:
Could this be `if (!unusedPath?.length) {}`?

Reply:
Ahh clever, yeah I think that should work!

Reply:
Fix in a138965